src/audio/mux/mux.c

```diff
@@ -82,15 +83,17 @@ static int mux_set_values(struct comp_dev *dev, struct comp_data *cd,
 	}
 
 	if (dev->comp.type == SOF_COMP_MUX) {
-		for (j = 0 ; j < PLATFORM_MAX_CHANNELS; j++) {
-			channel_set = false;
-			for (i = 0 ; i < cfg->num_streams; i++) {
-				if (popcount(cfg->streams[i].mask[j])) {
-					if (!channel_set) {
-						channel_set = true;
-					} else {
-						comp_cl_err(&comp_mux, "mux_set_values(): mux component is not able to mix channels");
-						return -EINVAL;
+		for (i = 0 ; i < PLATFORM_MAX_CHANNELS; i++) {
```

Comment:
Can we have a comment explaining this check.

Reply:
added something...